src/main/java/com/aws/greengrass/secretmanager/LocalStoreMap.java

```diff
@@ -0,0 +1,303 @@
+        }
+
+        return GetSecretValueResponse.builder().secretString(decryptedSecretString).secretBinary(decryptedSecretBinary)
+                .name(awsSecretResponse.getName()).arn(awsSecretResponse.getArn())
+                .createdDate(Instant.ofEpochMilli(awsSecretResponse.getCreatedDate()))
+                .versionId(awsSecretResponse.getVersionId()).versionStages(awsSecretResponse.getVersionStages())
+                .build();
+    }
+
+    /**
+     * Store secrets map.
+     * For the given secrets manager configuration
+     * {
+     * "cloudSecrets":[
+     *     {"arn":"secret1-arn", "labels":["AWSCURRENT", "NEW"]},
+     *     {"arn":"secret2-arn", "labels":["AWSCURRENT"]}
+     *   ]
+     * }
+     * Local store (runtime config) contains secrets in the following format
+     * [secrets:
+     * {"name": "secret1", "arn": "secret1-arn", "versionId": "secret1-v1", versionStages:["AWSCURRENT"]},
+     * {"name": "secret1", "arn": "secret1-arn", "versionId": "secret1-v2", versionStages:["NEW", "LATEST"]},
+     * {"name": "secret2", "arn": "secret2-arn", "versionId": "secret2-v1", versionStages:["AWSCURRENT"]}
+     * ]
+     * The following method returns the following structure for the same secrets only when decrypting the secret is
+     * successful
+     * {
+     * "secret1-arn" :{
+     *  "AWSCURRENT":{"name": "secret1", "arn": "secret1-arn", "versionId": "secret1-v1", versionStages:["AWSCURRENT"]},
+     *  "NEW" : {"name": "secret1", "arn": "secret1-arn", "versionId": "secret1-v2", versionStages:["NEW", "LATEST"]},
+     *  "LATEST": {"name": "secret1", "arn": "secret1-arn", "versionId": "secret1-v2", versionStages:["NEW", "LATEST"]}
+     * }
+     * "secret2-arn" : {
+     *   "AWSCURRENT":{"name": "secret2", "arn": "secret2-arn", "versionId": "secret2-v1", versionStages:["AWSCURRENT"]}
+     *  }
+     * }
+     */
+    private void reloadSecretsFromLocalStore() {
+        try {
+            SecretDocument doc = secretStore.getAll();
+            if (doc == null || doc.getSecrets() == null || doc.getSecrets().isEmpty()) {
+                return;
+            }
+            for (AWSSecretResponse secretResponse : doc.getSecrets()) {
+                try {
+                    decrypt(secretResponse);
+                } catch (SecretCryptoException e) {
+                    logger.atInfo().kv("secretArn", secretResponse.getArn()).kv("label", secretResponse.getVersionId())
+                            .log("Unable to decrypt the secret in local store.");
+                    continue;
+                }
+                secrets.putIfAbsent(secretResponse.getArn(), new Labels(new HashMap<>()));
+                secretResponse.getVersionStages().forEach((label) -> {
+                    secrets.get(secretResponse.getArn()).responseMap.put(label, secretResponse);
+                });
+            }
+        } catch (SecretManagerException e) {
```

Comment:
small: we can better scope the end of the try block to only include line 274 `SecretDocument doc = secretStore.getAll();` , and then add in an explicit return in this catch block. That way this catch is more co-located with the appropriate line and doesn't nest the rest of the logic inside of the try